modules/p2/ding_entity_rating_loan_history/ding_entity_rating_loan_history.module

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/**
+ * @file
+ * Implement ding_interaction plugin to display recent loans for rating.
+ */
+
+define('DING_ENTITY_RATING_LOAN_HISTORY_MAX_AMOUNT', 5);
+
+/**
+ * Implements hook_ding_interaction_info().
+ */
+function ding_entity_rating_loan_history_ding_interaction_info() {
+  $active = (user_is_logged_in() && ding_user_consent_has_consent('loan_history_store'));
```

Comment:
As far as I can tell there is no reason the check if the user is logged in. `ding_user_consent_has_consent()` handles all of this.